libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -314,3 +321,77 @@ def test_that_multiple_vector_stores_can_be_created(
     # Check that the name of the table being created for the embeddingstore
     # is what is expected.
     assert new_store._embedding_store.__table__.name == "langchain_vector_store_tests_2"
+
+    # Drop the new_store table to clean up this test run.
+    new_store.drop()
+
+
+def test_that_schema_input_is_used() -> None:
+    """Tests that when a schema is given as input to the SQLServer_VectorStore object,
+    a vector store is created within the schema."""
+    connection = create_schema()
+
+    # Create a vector store in the DB with the schema just created
+    sqlserver_vectorstore = SQLServer_VectorStore(
+        connection=connection,
+        connection_string=_CONNECTION_STRING,
+        db_schema=_SCHEMA,
+        embedding_function=FakeEmbeddings(size=1536),
+        table_name=_TABLE_NAME,
+    )
+    sqlserver_vectorstore.add_texts(["cats"])
+
+    # Confirm table in that schema exists.
+    result = connection.execute(text(_SYS_TABLE_QUERY % (_TABLE_NAME, _SCHEMA)))
+    assert result.fetchone() is not None, _TABLE_DOES_NOT_EXIST % (_SCHEMA, _TABLE_NAME)
+    connection.close()
+
+
+def test_that_same_name_vector_store_can_be_created_in_different_schemas() -> None:
+    """Tests that vector stores can be created with same name in different
+    schemas even with the same connection."""
+    connection = create_schema()
+
+    # Create a vector store in the DB with the schema just created
+    sqlserver_vectorstore = SQLServer_VectorStore(
+        connection=connection,
+        connection_string=_CONNECTION_STRING,
+        db_schema=_SCHEMA,
+        embedding_function=FakeEmbeddings(size=1536),
+        table_name=_TABLE_NAME,
+    )
+
+    # Create a vector store in the DB with the default schema
+    sqlserver_vectorstore_default_schema = SQLServer_VectorStore(
+        connection=connection,
+        connection_string=_CONNECTION_STRING,
+        embedding_function=FakeEmbeddings(size=1536),
+        table_name=_TABLE_NAME,
+    )
+
+    sqlserver_vectorstore.add_texts(["cats"])
+    result_with_schema = connection.execute(
+        text(_SYS_TABLE_QUERY % (_TABLE_NAME, _SCHEMA))
+    )
+    assert result_with_schema.fetchone() is not None, _TABLE_DOES_NOT_EXIST % (
+        _SCHEMA,
+        _TABLE_NAME,
+    )
+
+    sqlserver_vectorstore_default_schema.add_texts(["cats"])
+    result_with_default = connection.execute(
+        text(_SYS_TABLE_QUERY % (_TABLE_NAME, "dbo"))
+    )
+    assert result_with_default.fetchone() is not None, _TABLE_DOES_NOT_EXIST % (
+        "dbo",
+        _TABLE_NAME,
+    )
+    connection.close()
+
+
+def create_schema() -> Connection:
```

Comment:
The connection returned by create_schema is not a connection specific to the schema that you created here. So, doesnt make sense to return that. I would imagine create_schema to just do that using a connection as input.

Reply:
Noted. Will modify this in the next PR.